File: 240331/개발팀의 능력/the-capabilities-of-the-development-team.java
import java.util.*;
import java.io.*;

public class Main {
    public static int[] developers = new int[5];
    public static int minDiff = Integer.MAX_VALUE;
    public static int sum;
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        for (int i = 0; i < developers.length; i++) {
            developers[i] = sc.nextInt();
        }
        sum = Arrays.stream(developers).sum();

        for (int i = 0; i < developers.length; i++) {
            for (int j = i+1; j < developers.length; j++) {
                for (int k = 0; k < developers.length; k++) {
                    for (int l = k+1; l < developers.length; l++) {
                        if (k == i || k == j || l == i || l == j) {
                            continue;
                        }
                        int team1 = developers[i] + developers[j];
                        int team2 = developers[k] + developers[l];
                        int team3 = sum - team1 - team2;
                        if (team1 == team2 || team2 == team3 || team1 == team3) {
                            continue;
                        }
                        minDiff = Math.min(minDiff, getMinDiff(team1, team2, team3));
                    }
                }
            }
        }
        if (minDiff == Integer.MAX_VALUE) {
            System.out.println(-1);
        } else {
            System.out.println(minDiff);
        }
    }

    public static int getMinDiff(int num1, int num2, int num3) {
        int result = Math.abs(num1 - num2);
        result = Math.max(result, Math.abs(num2 - num3));
        result = Math.max(result, Math.abs(num1 - num3));
        return result;
    }
}

File: 240331/개발팀의 능력/the-capabilities-of-the-development-team.py
import sys

developers = list(map(int, input().split()))
total_sum = sum(developers)
min_diff = sys.maxsize

def get_diff(i, j, k):
    return_value = sys.maxsize

    team1 = developers[i]
    team2 = developers[j] + developers[k]
    team3 = total_sum - team1 - team2

    if (team1 == team2 or team2 == team3 or team1 == team3):
        return return_value
    
    ret = abs(team1 - team2)
    ret = max(ret, abs(team2 - team3))
    ret = max(ret, abs(team3 - team1))

    return ret

for i in range(len(developers)):
    for j in range(i+1, len(developers)):
        for k in range(j+1, len(developers)):
            min_diff = min(min_diff, get_diff(i, j, k))

if (min_diff == sys.maxsize):
    print(-1)
else:
    print(min_diff)